File: BookTracker.Api/Entities/Book.cs
namespace BookTracker.Api.Entities;

public class Book
{
    public long Id { get; set; }
    public string Title { get; set; } = null;
    public string Author { get; set; } = null;
    public string Isbn { get; set; } = null;
    public int Pages { get; set; }
    public int Rating { get; set; }
    public DateTime CreatedAt { get; set; } =  DateTime.UtcNow;
}

File: BookTracker.Ui/src/types/book.ts
export type Book = {
  id: number;
  title: string;
  author: string;
  isbn: string;
  pages: number;
  rating: number;
  createdAt?: Date;
}
export type PaginatedResponse<T> = {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
}

export type BookQueryParams = {
  page?: number;
  pageSize?: number;
  search?: string;
}

export type CreateBookDto = {
  title: string;
  author: string;
  isbn: string;
  pages: number;
  rating: number;
}
